log-viewer/src/main/java/com/logviewer/data2/Log.java

```diff
@@ -141,6 +147,25 @@ public Snapshot createSnapshot() {
         }
     }
 
+    private void decompressAndCopyGZipFile() throws IOException {
+
+        File tempFile = File.createTempFile("log-viewer-", "-" + file.getName(file.getNameCount() - 1) + ".tmp");
```

Comment:
I suggest to change it to NIO2

```java
Path tempFile = Files.createTempFile("log-viewer-", "-" + file.getName(file.getNameCount() - 1) + ".tmp");
Files.deleteIfExists(tempFile);
```

Reply:
I did some searches, and find out there's no direct way in java nio API to get from tgz to SeekableByteChannel. 
I have to introduce some external dependencies and/or write custom code to accomplish this. 

I appreciate your help on this.